src/util/newconfig/ConfigDefinition.cpp

```diff
@@ -101,113 +108,102 @@ static ClioConfigDefinition ClioConfig = ClioConfigDefinition{
      {"api_version.max", ConfigValue{ConfigType::Integer}}}
 };
 
+ClioConfigDefinition::ClioConfigDefinition(std::initializer_list<KeyValuePair> pair)
+{
+    for (auto const& p : pair) {
```

Comment:
You should be able to make this slightly nicer by using `auto const& [key, value] : pair` and use the two variables below instead of `first` and `second`.